### PR TITLE
feat(telecom): add delete modal with cta

### PR DESCRIPTION
**Pull request by antonymarion**
ref: MANAGER-13975

| Question         | Answer
| ---------------- | ---
| Branch?          | `master` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Feat#MANAGER-13975
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
~~- [ ] Breaking change is mentioned in relevant commits~~

## Description

Add modal for cta delete device (Softphone).

Notice: Some data has been mocked since api not available yet.

### CODE REVIEW

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js [hunk 1]
```diff
@@ -1,9 +1,20 @@
 import { MOBILE_OS } from './softphone.constants';
 
 export default class SoftphoneController {
+  /* @ngInject */
+  constructor($state) {
+    this.$state = $state;
+  }
```
Comment (darsene) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js:7]: Should be defined in the routing file rather
```suggestion
  constructor($state) {
    this.$state = $state;
  }
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -9,6 +9,13 @@
   "telephony_line_softphone_download_links_title": "Liens de téléchargements",
   "telephony_line_softphone_configure_title": "Configurer la ligne",
   "telephony_line_softphone_configure_description": "Vous avez la possibilité de configurer jusqu'à trois appareils au total. Vous pouvez configurer un seul appareil à la fois.",
+  "telephony_line_softphone_delete_device_error": "La suppression de votre softphone a échoué: {{status}}.",
+  "telephony_line_softphone_btn_delete": "Supprimer",
+  "telephony_line_softphone_btn_rerecord": "Ré-enregistrer",
+  "telephony_line_softphone_delete_device_modal_title": "Supression de mon appareil",
+  "telephony_line_softphone_delete_device_warning": "Êtes-vous sûr(e) de vouloir supprimer cet appareil de votre application Softphone ?",
+  "telephony_line_softphone_delete_device_before_action": "Cette action entraînera l'impossibilitié d'utiliser cet appareil pour passer ou recevoir des appels via l'application. Toute configuration ou attribution spécifique liée à cet appareil sera également perdue.",
```
Comment (darsene) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/translations/Messages_fr_FR.json:17]:
```suggestion
  "telephony_line_softphone_delete_device_before_action": "Cette action entraînera l'impossibilité d'utiliser cet appareil pour passer ou recevoir des appels via l'application. Toute configuration ou attribution spécifique liée à cet appareil sera également perdue.",
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/deleteDevice/delete-device.controller.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+export default class DeleteDeviceController {
+  /* @ngInject */
+  constructor($translate, $stateParams, softphoneService, TucToast) {
+    this.$translate = $translate;
+    this.TucToast = TucToast;
+    this.$stateParams = $stateParams;
+    this.softphoneService = softphoneService;
+  }
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/deleteDevice/delete-device.controller.js:8]: **nitpick:** Dont inject `$stateParams` in controller but add an input to the component and send it from routing file.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js [hunk 4]
```diff
@@ -24,4 +35,11 @@ export default class SoftphoneController {
     }
     this.osName = MOBILE_OS.windows;
   }
+
+  deleteDevice(deviceId) {
+    this.$state.go(
+      'telecom.telephony.billingAccount.line.dashboard.softphone.delete',
+      { deviceId },
+    );
+  }
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.controller.js:44]: Add a `goToDeleteDevice` function in the routing. Dont inject `$state` into controller.

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 5]
```diff
@@ -72,7 +74,53 @@ <h4 class="pt-3">
...
+                </oui-guide-menu-item>
+            </oui-guide-menu>
+        </div>
+        <div
+            class="d-flex flex-wrap w-100 minw-0 align-items-center justify-content-start"
+        >
+            <oui-tile
+                data-ng-repeat="deviceId in ::$ctrl.softphoneDevices track by $index"
+                class="m-2"
+                heading="{{ 'Paul Iphone' }}"
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:96]: **nitpick:** Dont we need translations for this?

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html [hunk 6]
```diff
@@ -72,7 +74,53 @@ <h4 class="pt-3">
...
+            >
+                <oui-tile-definition term="Statut" description="Connecté">
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:98]: **nitpick:** Dont we need translations for this?

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html (hunk 5 above)
Comment (antonymarion) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.html:96]: Same as the other one

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/translations/Messages_fr_FR.json (hunk 2 above)
Comment (antonymarion) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/translations/Messages_fr_FR.json:17]: thks

**Review by darsene — APPROVED** [review]

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/deleteDevice/delete-device.routing.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        },
+      },
+      layout: 'modal',
+      resolve: {
+        goBack: /* @ngInject */ ($state) => (reload) =>
+          $state.go('^', {}, { reload }),
+        breadcrumb: () => null,
+        billingAccount: ($stateParams) => $stateParams.billingAccount,
+        serviceName: ($stateParams) => $stateParams.serviceName,
+        deviceId: ($stateParams) => $stateParams.deviceId,
```
Comment (anooparveti) [inline packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/deleteDevice/delete-device.routing.js:18]:
```suggestion
        billingAccount: /* @ngInject */ ($stateParams) => $stateParams.billingAccount,
        serviceName: /* @ngInject */ ($stateParams) => $stateParams.serviceName,
        deviceId: /* @ngInject */ ($stateParams) => $stateParams.deviceId,
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/deleteDevice/delete-device.routing.js (hunk 7 above)
Comment (antonymarion) [reply packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/deleteDevice/delete-device.routing.js:18]: oh dear.thks :)

**Review by anooparveti — APPROVED** [review]

**Pull request merged**